doc/Building_targets.md

```diff
@@ -0,0 +1,179 @@
+# Building targets
+
+KOReader is available for multiple platforms. Here are instructions to build installable packages for all these platforms.
+
+These instructions are intended for a Linux OS. MacOS and Windows users are suggested to develop in a Linux VM.
+
+
+## Prerequisites
+
+This instructions asume that you [have a development environment ready to run](Building.md) KOReader. If not then please install common prerequisites first.
+
+### A toolchain for your target.
+
+Each target has its own architecture and you'll need to setup a proper cross-compile toolchain. Your GCC should be at least version 4.8
+
+#### for Android
+
+A compatible version of the Android NDK and SDK will be downloaded automatically by `./kodev release android` if no NDK or SDK is provided in environment variables. For that purpose you can use:
+
+```
+NDK=/ndk/location SDK=/sdk/location ./kodev release android
+```
+
+If you want to use your own installed tools please make sure that you have the **NDKr15c** and the SDK for Android 9 (**api level 28**) already installed.
+
+#### for embedded linux devices
+
+Cross compile toolchains are available for Ubuntu users through these commands:
+
+##### Kindle and Cervantes
+
+```
+sudo apt-get install gcc-arm-linux-gnueabi g++-arm-linux-gnueabi
+```
+
+##### Kobo and Ubuntu Touch
+
+```
+sudo apt-get install gcc-arm-linux-gnueabihf g++-arm-linux-gnueabihf
+```
+
+**NOTE 1:** The packages `pkg-config-arm-linux-gnueabihf` and `pkg-config-arm-linux-gnueabi` may
+block you from building. Remove them if you get the ld error
+
+```
+/usr/lib/gcc-cross/arm-linux-gnueabihf/4.8/../../../../arm-linux-gnueabihf/bin/ld: cannot find -lglib-2.0
+```
+
+**NOTE 2:** In the specific case of Cervantes, Kindle & Kobo targets, while we make some effort to support these Linaro/Ubuntu TCs,
+they do *not* exactly target the proper devices. While your build will go fine, this may lead to runtime failure.
+As time goes by, and/or the more bleeding-edge your distro is, the greater the risk for mismatch gets.
+Thankfully, we have a distribution-agnostic solution for you: [koxtoolchain](https://github.com/koreader/koxtoolchain)!
+This will allow you to build the *exact* same TCs used to build the nightlies, thanks to the magic of [crosstool-ng](https://github.com/crosstool-ng/crosstool-ng).
```

Comment:
```suggestion
This will allow you to build the *exact* same TCs used to build the nightlies, thanks to the magic of [crosstool-ng](https://github.com/crosstool-ng/crosstool-ng). These are also included precompiled in the Docker images for the respective targets.
```